lib/src/chargebee.dart

```diff
@@ -31,231 +28,128 @@ class Chargebee {
-        print('result : $result');
-        List<dynamic> jsonData = jsonDecode(result.toString());
-        for (var value in jsonData) {
-          var wrapper = SubscriptonList.fromJson(value);
-          subscriptions.add(wrapper.subscripton!);
+        List jsonData = jsonDecode(result.toString());
+        if(jsonData.isNotEmpty) {
+          for (var value in jsonData) {
+            var wrapper = SubscriptonList.fromJson(value);
+            subscriptions.add(wrapper.subscripton!);
+          }
         }
-        print(subscriptions.first.subscriptionId);
-        print(subscriptions.first.status);
-        return subscriptions;
-      } on CBException catch (e) {
-        log('CBException : ${e.message}');
-      }
     } else {
-      try {
         String result = await platform.invokeMethod(
             Constants.mSubscriptionMethod, queryParams);
-        log('result : $result');
-
-        List<dynamic> jsonData = jsonDecode(result);
-        for (var value in jsonData) {
-          var wrapper = SubscriptonList.fromJsonAndroid(value);
-          subscriptions.add(wrapper.subscripton!);
+        List jsonData = jsonDecode(result);
+        if(jsonData.isNotEmpty) {
+          for (var value in jsonData) {
+            var wrapper = SubscriptonList.fromJsonAndroid(value);
+            subscriptions.add(wrapper.subscripton!);
+          }
         }
-        print(subscriptions.first.subscriptionId);
-        print(subscriptions.first.status);
-        return subscriptions;
-      } on CBException catch (e) {
-        log('CBException : ${e.message}');
-      }
     }
     return subscriptions;
   }
 
   /* Get Apple/Google Product ID's from chargebee system */
-  static Future<List<dynamic>> retrieveProductIdentifers(
-      Map<String, String> queryParams) async {
-    List<dynamic> productIdList = [];
-
-    if (Platform.isIOS) {
-      try {
-        String result = await platform.invokeMethod(
-            Constants.mProductIdentifiers, queryParams);
-        log('result : $result');
-        productIdList = jsonDecode(result);
-
-        return productIdList;
-      } on CBException catch (e) {
-        log('CBException : ${e.message}');
-      }
-    } else {
-      try {
-        String result = await platform.invokeMethod(
-            Constants.mProductIdentifiers, queryParams);
-        print("result : $result");
-        productIdList = jsonDecode(result);
-
-        return productIdList;
-      } on CBException catch (e) {
-        log('CBException : ${e.message}');
-      }
-    }
-    return productIdList;
+  static Future<List> retrieveProductIdentifers(
+      [Map<String, String>? queryParams]) async {
+      String result = await platform.invokeMethod(
+          Constants.mProductIdentifiers, queryParams);
+      return jsonDecode(result);
   }
 
   /* Get entitlement details from chargebee system */
-  static Future<List<dynamic>> retrieveEntitlements(
+  static Future<List> retrieveEntitlements(
```

Comment:
We should also define the type of this `List`, but I think we can take it up separately.

Reply:
Yes, we can do it as part of improvements task.